### DISCUSSION TITLE
Eclipselink 3.x and 4.x compatibility with Spring Boot 2.x / Spring Data?

### DISCUSSION THREAD

**Discussion by wilx** [eclipse-ee4j/eclipselink#1744] [Q&A] [answered]
We are currently using Eclipselink 2.7.x with Spring Boot 2.6.x. Does Eclipselink 3.x and/or 4.x claim any compatibility with any version of Spring Boot and Spring Data? Has anyone tried it?

What are the major differences between 2.7.x and 3.x and 4.x that they are each different major version?

**Comment by lukasj** [comment, marked as answer]
> Does Eclipselink 3.x and/or 4.x claim any compatibility with any version of Spring Boot and Spring Data? Has anyone tried it?

No, I'm not aware of that

> What are the major differences between 2.7.x and 3.x and 4.x that they are each different major version?

2.7.x is for Java SE 8 + Java EE 8/Jakarta EE 8 -> `javax.persistence` package namespace
3.0.x is for Java SE 8 + Jakarta EE 9 -> `jakarta.persistence` package namespace
4.0.x is for Java SE 11 & Jakarta EE 10, contains few breaking changes in some EclipseLink APIs, like ie removal of deprecated methods, renaming, etc
